docs/async.md

```diff
@@ -53,6 +53,9 @@ async with httpx.AsyncClient() as client:
     ...
 ```
 
+!!! warning
+Note that there is some overhead to instantiating and terminating the client object, so it's recommended to avoid building the client using `async with` repeatedly inside a "hot loop". This can be achieved by having a single global client instance which will provide automatic connection pooling.
```

Comment:
I've made a suggestion based on your first pass, but with a positive phrasing...

```suggestion
In order to get the most benefit from connection pooling, make sure you're not instantiating multiple client instances. This can be achieved either by having a single scoped client that's passed throughout wherever it's needed, or by having a single global client instance.
```

What do you think?

Reply:
Hm I think it's really import to be explicit about not using async with in hot loops. Thinking back to when I started with python async, the suggested text here would probably not be enough for me to understand clearly that I was going against recommended practices. How about:

```
In order to get the most benefit from connection pooling, make sure you're not instantiating multiple client instances - for example by using `async with` repeatedly inside a "hot loop".
```
Best of both worlds?